examples/decoding/plot_ssd_spatial_filters.py

```diff
@@ -0,0 +1,145 @@
+"""
+===========================================================
+Compute Sepctro-Spatial Decomposition (SDD) spatial filters
+===========================================================
+
+In this example, we will compute spatial filters for retaining
+oscillatory brain activity and down-weighting 1/f background signals
+as proposed by [1]_.
+The idea is to learn spatial filters that separate oscillatory dynamics
+from surrounding non-oscillatory noise based on the covariance in the
+frequency band of interest and the noise covariance absed on surrounding
+frequencies.
+
+References
+----------
+
+.. [1] Nikulin, V. V., Nolte, G., & Curio, G. (2011). A novel method for
+       reliable and fast extraction of neuronal EEG/MEG oscillations on the
+       basis of spatio-spectral decomposition. NeuroImage, 55(4), 1528-1535.
+"""
+# Author: Denis A. Engemann <denis.engemann@gmail.com>
+#
+# License: BSD (3-clause)
+
+import matplotlib.pyplot as plt
+from scipy.linalg import eigh
+import numpy as np
+import mne
+from mne import Epochs
+from mne.datasets.fieldtrip_cmc import data_path
+from mne.utils import _time_mask
+from mne.channels import read_layout
+
+
+def freq_mask(freqs, fmin, fmax):
+    """convenience function to select frequencies"""
+    return _time_mask(freqs, fmin, fmax)
+
+# Define parameters
+fname = data_path() + '/SubjectCMC.ds'
+raw = mne.io.read_raw_ctf(fname)
+raw.crop(50., 250.).load_data()  # crop for memory purposes
+
+freqs_sig = 9, 12
+freqs_noise = 8, 13
+
+# Filter MEG data to focus on beta band
+raw.pick_types(meg=True, ref_meg=True, eeg=False, eog=False)
+raw_sig = raw.copy()
+raw_sig.filter(*freqs_sig, l_trans_bandwidth=1, h_trans_bandwidth=1,
+               fir_design='firwin')
+
+raw_noise = raw.copy()
+raw_noise.filter(*freqs_noise, l_trans_bandwidth=1, h_trans_bandwidth=1,
+                 fir_design='firwin')
+
+# remove signal from noise to further increase impact of surrounding freqs
+raw_noise._data -= raw_sig._data
+
+# Build epochs as sliding windows over the continuous raw file
+events = mne.make_fixed_length_events(raw, id=1, duration=.250)
+
+# Epoch length is 1.5 second
+picks = mne.pick_types(raw.info, meg=True, ref_meg=False)
+epochs_sig = Epochs(raw_sig, events, tmin=0., tmax=1.500, baseline=None,
+                    picks=picks, detrend=1, decim=4)
```

Comment:
detrending actually helps? I am surprised after doing some high pass filtering

Reply:
no it's useless, actually did not intend to copy that.